core/src/main/scala/com/mesosphere/usi/core/javaapi/SchedulerAdapter.scala

```diff
@@ -0,0 +1,83 @@
+package com.mesosphere.usi.core.javaapi
+
+import java.util.Optional
+import java.util.concurrent.{CompletableFuture, CompletionStage}
+
+import akka.{Done, NotUsed}
+import akka.stream.{Materializer, OverflowStrategy, QueueOfferResult, scaladsl}
+import akka.stream.javadsl.{Sink, SinkQueueWithCancel, Source, SourceQueueWithComplete}
+import com.mesosphere.usi.core.Scheduler.{SpecInput, StateOutput}
+import com.mesosphere.usi.core.scalaapi.{SchedulerAdapter => ScalaSchedulerAdapter}
+import com.mesosphere.usi.core.models.{SpecUpdated, SpecsSnapshot, StateEvent, StateSnapshot}
+
+import scala.compat.java8.FutureConverters._
+import akka.japi.tuple.Tuple3
+
+import scala.concurrent.ExecutionContext
+
+class SchedulerAdapter(
+    schedulerFlow: scaladsl.Flow[SpecInput, StateOutput, NotUsed],
+    mat: Materializer,
+    ec: ExecutionContext) {
+
+  private implicit val materializer: Materializer = mat
+  private implicit val executionContext: ExecutionContext = ec
+
+  private val delegate = new ScalaSchedulerAdapter(schedulerFlow)
+
+  /**
+    * Represents the scheduler as a Sink and Source.
+    *
+    * This method will materialize the scheduler first, then Sink and Source can be materialized independently.
+    *
+    * @param specsSnapshot Snapshot of the current specs
+    * @return Snapshot of the current state, as well as Source which produces StateEvents and Sink which accepts SpecEvents
+    */
+  def asSourceAndSink(specsSnapshot: SpecsSnapshot)
+    : Tuple3[CompletableFuture[StateSnapshot], Source[StateEvent, NotUsed], Sink[SpecUpdated, NotUsed]] = {
+    val (snap, source, sink) = delegate.asSourceAndSink(specsSnapshot)
+    Tuple3(snap.toJava.toCompletableFuture, source.asJava, sink.asJava)
+  }
+
+  def asSourceAndSink()
```

Comment:
I still don't understand why we explicitly support providing no snapshot. I think it's pretty dangerous :) I would rather all orchestration code had to explicitly provide the empty one when that's really what's needed...